src/index.js

```diff
@@ -300,18 +303,29 @@ class MongoOplogReader {
       });
   }
 
+  deleteOpMembers(opId) {
+    // wait a few seconds to prevent redundant workers from recreating the key immediately after it's deleted
+    // if a redundant worker is way slower, this key will be purged passively after the TTL
+    setTimeout(() => {
```

Comment:
I don't like the `setTimeout` solution because it assumes that whatever might be happening is done in `N seconds` and that may not always be true.

Are there any other solutions?

Food for thought: Would be nice if each `op` could set a "clean up" flag. 

(Maybe I am misunderstanding the entire purpose of the code, if so, please explain)

Reply:
Agreed, I like to avoid setTimeout. But the only alternative I can think of is to write a garbage collector that runs in the background--which is less effective/efficient.

Here's the problem, we have redundant workers receiving "oplog events". As soon as two members have seen the oplog event, we emit the "food event" and then we need to delete `members`.

Suppose workers receive the oplog event in this order:
```
worker  members
A1      [A]
B1      [A,B]  --> we have 2 members! emit the food event and delete members
A2      [A]
B2      [A,B]  --> we have 2 members! (try to) emit the food event and delete members
```
In this case we're all good because `members` has been deleted.

But suppose workers receive the oplog event in this order:
```
worker  members
A1      [A]
A2      [A]
B1      [A,B]  --> we have 2 members! emit the event and delete members
B2      [B]
```

You can see B2 fires a few milliseconds after `members` is deleted and recreates `members`. 

The obvious fix is `setTimeout` but I'm open to alternatives. I agree `N seconds` is arbitrary but I set it high enough that I'm not concerned about the outliers that may not get deleted.